NDTensors/test/runtests.jl

```diff
@@ -1,43 +1,25 @@
-using Test
-using SafeTestsets
-
-println("Passing arguments ARGS=$(ARGS) to test.")
-
-if isempty(ARGS) || "base" in ARGS
-  println(
-    """\nArguments ARGS = $(ARGS) are empty, or contain `"base"`. Running cpu NDTensors tests.""",
-  )
-end
-if "cuda" in ARGS || "all" in ARGS
-  println("""\nArguments ARGS = $(ARGS) contain `"cuda"`. Running NDTensorCUDA tests.""")
-  using CUDA
-end
-if "metal" in ARGS || "all" in ARGS
-  println("""\nArguments ARGS = $(ARGS) contain`"metal"`. Running NDTensorMetal tests.""")
-  using Metal
-end
+using SafeTestsets: @safetestset
 
 @safetestset "NDTensors" begin
-  @testset "$filename" for filename in [
-    "lib/runtests.jl",
-    "linearalgebra.jl",
-    "dense.jl",
-    "blocksparse.jl",
-    "diagblocksparse.jl",
-    "diag.jl",
-    "emptynumber.jl",
-    "emptystorage.jl",
-    "combiner.jl",
-    "arraytensor/arraytensor.jl",
-    "ITensors/runtests.jl",
-  ]
-    println("Running $filename")
-    include(filename)
+  using Test: @testset
+  @testset "$(@__DIR__)" begin
+    filenames = filter(readdir(@__DIR__)) do f
+      startswith("test_")(f) && endswith(".jl")(f)
+    end
+    for dir in ["lib/", "arraytensor/", "ITensors/"]
+      push!(filenames, dir * "runtests.jl")
+    end
```

Comment:
```suggestion
    for dir in ["lib", "arraytensor", "ITensors"]
      push!(filenames, joinpath(dir, "runtests.jl"))
    end
```
`joinpath("a", "b")` output `"a/b"`, but also handles creating paths cross-platform (i.e. handles Windows conventions for how paths are defined).

Reply:
Ah yes that is very important, sorry and thank you for the reminder!